pkg/operator/configobserver/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,150 @@
+		return existingConfig, append(errs, err)
+	}
+
+	observedAccessTokenMaxAgeSeconds := float64(oauthConfig.Spec.TokenConfig.AccessTokenMaxAgeSeconds)
+	if observedAccessTokenMaxAgeSeconds == 0 {
+		// As the value 0 indicates that this field is not set in OAuth cluster config, use the default value.
+		observedAccessTokenMaxAgeSeconds = defaultAccessTokenMaxAgeSeconds
+	}
+
+	if existingAccessTokenMaxAgeSeconds != observedAccessTokenMaxAgeSeconds {
+		recorder.Eventf("ObserveTokenConfig", "%s changed from %d to %d", fieldAccessTokenMaxAgeSeconds,
+			existingAccessTokenMaxAgeSeconds,
+			observedAccessTokenMaxAgeSeconds)
+	}
+
+	return buildUnstructuredTokenConfig(observedAccessTokenMaxAgeSeconds, tokenMaxAgePath), errs
+}
+
+// ObserveAccessTokenInactivityTimeout returns an unstructured fragment of KubeAPIServerConfig that has access token inactivity timeout,
+// if there is a valid value for it in OAuth cluster config.
+func ObserveAccessTokenInactivityTimeout(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	errs = []error{}
+	observedConfig := map[string]interface{}{}
+	tokenConfigPath := []string{"oauthConfig", "tokenConfig"}
+	tokenInactivityTimeoutPath := append(tokenConfigPath, fieldAccessTokenInactivityTimeoutSeconds)
+	defer func() {
+		// Prune the observed config so that it only contains access token inactivity timeout field.
+		ret = configobserver.Pruned(ret, tokenInactivityTimeoutPath)
+	}()
+
+	listers, ok := genericlisters.(OAuthLister)
+	if !ok {
+		return existingConfig, append(errs, fmt.Errorf("failed to assert: given lister does not implement OAuth lister"))
+	}
+
+	existingAccessTokenInactivityTimeout, _, err := unstructured.NestedFieldCopy(existingConfig, tokenInactivityTimeoutPath...)
+	if err != nil {
+		// If failed to read existing config for access token inactivity timeout, continue execution with the
+		// hope to find the value for it in OAuth cluster config.
+		errs = append(errs, err)
+	}
+
+	var existingAccessTokenInactivityTimeoutFloat float64
+	if existingAccessTokenInactivityTimeout != nil {
+		if f, ok := existingAccessTokenInactivityTimeout.(float64); ok {
+			existingAccessTokenInactivityTimeoutFloat = f
+		}
+	}
+
+	oauthConfig, err := listers.OAuthLister().Get("cluster")
+	if err != nil {
+		// Failed to read OAuth cluster config.
+		if errors.IsNotFound(err) {
+			klog.Warning("oauth.config.openshift.io/cluster: not found")
+		}
+
+		if existingAccessTokenInactivityTimeoutFloat == 0 {
+			// As the nil value indicates that an attempt to read from existing config failed previously, return empty config.
+			return observedConfig, append(errs, err)
+		}
+		// Return whatever is present in existing config
+		return existingConfig, append(errs, err)
+	}
+
+	observedAccessTokenInactivityTimeout := oauthConfig.Spec.TokenConfig.AccessTokenInactivityTimeout.Seconds()
```

Comment:
if the user sets `"3600.5s"` for the metav1.Duration, it will marshal into the same number in the kube apiserver config as float and the unmarshalling will fail in kube-apiserver because comma values do not work with an int32 in https://github.com/openshift/api/blob/11eee842dafd4698259c5de6b0746183ea6998f5/osin/v1/types.go#L393.

Reply:
Good catch! I am guessing this will also be resolved with new inactivity timeout field in osin config.

Reply:
yes

Reply:
as discussed in Slack, just round to the next integer. That should be enough. Sub-second durations are possible, but don't make much sense.